contracts/margin/external/ERC20/ERC20PositionWithdrawer.sol

```diff
@@ -0,0 +1,122 @@
+/*
+
+    Copyright 2018 dYdX Trading Inc.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+*/
+
+pragma solidity 0.4.24;
+pragma experimental "v0.5.0";
+
+import { WETH9 } from "canonical-weth/contracts/WETH9.sol";
+import { ERC20Position } from "./ERC20Position.sol";
+import { TokenInteract } from "../../../lib/TokenInteract.sol";
+import { ExchangeWrapper } from "../../interfaces/ExchangeWrapper.sol";
+
+
+/**
+ * @title ERC20PositionWithdrawer
+ * @author dYdX
+ *
+ * Proxy contract to withdraw from an ERC20Position and exchange the withdrawn tokens on a DEX
+ */
+contract ERC20PositionWithdrawer
+{
+    using TokenInteract for address;
+
+    // ============ Constants ============
+
+    address public WETH;
+
+    // ============ Constructor ============
+
+    constructor(
+        address weth
+    )
+        public
+    {
+        WETH = weth;
+    }
+
+    // ============ Public Functions ============
+
+    /**
+     * Fallback function. Disallows ether to be sent to this contract without data except when
+     * unwrapping WETH.
+     */
+    function ()
+        external
+        payable
+    {
+        require( // coverage-disable-line
+            msg.sender == WETH,
+            "PayableMarginMinter#fallback: Cannot recieve ETH directly unless unwrapping WETH"
+        );
+    }
+
+    /**
+     * [withdrawFromERC20Position description]
+     *
+     * @param  erc20Position    The address of the ERC20Position contract to withdraw from
+     * @param  withdrawnToken   The address of the token to withdraw from the ERC20Position
+     * @param  returnedToken    The address of the token that is returned to the user
+     * @param  exchangeWrapper  The address of the ExchangeWrapper
+     * @param  orderData        Arbitrary bytes data for any information to pass to the exchange
+     * @return                  [1] The number of tokens withdrawn
+     *                          [2] The number of tokens returned to the user
+     */
+    function withdrawFromERC20Position(
+        address erc20Position,
+        address withdrawnToken,
+        address returnedToken,
+        address exchangeWrapper,
+        bytes orderData
+    )
+        external
+        returns (uint256, uint256)
+    {
+        // withdraw tokens
+        uint256 tokensWithdrawn = ERC20Position(erc20Position).withdraw(msg.sender);
+
+        // error checking
+        require(
```

Comment:
I think you should just return in this case, not throw an error. It's possible the user could do this multiple times